woocommerce/woocommerce-android
# Edit variation attributes

Summary
==========
Implements issues #3492, #3494 and partially #3210. This PR makes it possible to enter a Variation, access and edit all attributes with options defined by the parent Product and update the Variation to the site. Any change done by the `Edit Attributes` view flow inside the Variation List should reflect in the Variation attributes list.

Screenshots
==========
| Variation Details with Attributes Button  | Attributes list with possible options | Dialog with possible options | Updated Attributes list | Updated Variation Details View with Update button | Changes submission |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| ![screenshot-1616724080241](https://user-images.githubusercontent.com/5920403/112567049-e62ddc80-8dbe-11eb-84b7-d418063e9f6e.jpg) | ![screenshot-1616724084491](https://user-images.githubusercontent.com/5920403/112567064-ee861780-8dbe-11eb-8038-b6817043b6c1.jpg) | ![screenshot-1616724097171](https://user-images.githubusercontent.com/5920403/112567074-f34acb80-8dbe-11eb-9af9-dd6c299618b6.jpg) | ![screenshot-1616724110018](https://user-images.githubusercontent.com/5920403/112567085-f776e900-8dbe-11eb-84f6-24c73d3692ef.jpg) |   ![screenshot-1616724114630](https://user-images.githubusercontent.com/5920403/112567096-ffcf2400-8dbe-11eb-972d-7097ef5a8916.jpg) | ![screenshot-1616724120503](https://user-images.githubusercontent.com/5920403/112567104-0362ab00-8dbf-11eb-846d-7c0ccd52776e.jpg) |

How to Test
==========
1. Open a configured Variation from a Variable Product
2. Verify that all possible attribute for a given Variation is visible and selectable
3. Change any Attribute options
4. Verify that all selections and changes are correctly set
5. Return to the Variation Details View and verify that the Attributes button is updated correctly
6. Click on the Update button and verify that all changes were submitted

Update release notes:
